game/src/main/scala/io/aigar/game/Virus.scala

```diff
@@ -0,0 +1,48 @@
+package io.aigar.game
+
+import com.github.jpbetz.subspace.Vector2
+import io.aigar.game.serializable.Position
+
+object Virus {
+  final val Quantity = 15
+  final val Mass = 100
+}
+
+class Virus(var position: Vector2) {
+
+  def update(grid: Grid, players: List[Player]): Unit = {
+    if(detectCollisions(players)) {
+      // TODO Add the virus consumption into the cell
+      respawn(grid, players)
+    }
+  }
+
+  def detectCollisions(players: List[Player]): Boolean ={
+    for(player <- players) {
+      for(cell <- player.cells) {
+        // TODO Change the 1.1 value to the constant
+        if(cell.contains(position) && cell.mass > Virus.Mass * 1.1){
+          return true
+        }
+      }
+    }
+    false
+  }
+
+  def respawn(grid: Grid, players: List[Player]): Unit = {
+    val newPosition = grid.randomPosition
+
+    // We always want 15 viruses. How to we manage it without doing a while true loop?
+    for (player <- players) {
+      for (cell <- player.cells) {
+        if (cell.contains(newPosition)) return
+      }
+    }
+
+    position = newPosition
+  }
+
+  def state: Position = {
+    Vector2Utils.StateAddon(position).state
```

Comment:
It works, but what I mean is that you can just do `position.state` and it will work if `Vector2Utils.StateAddon` is imported. Look at [how we're doing it in `Cell`](https://github.com/DrPandemic/aigar.io/blob/master/game/src/main/scala/io/aigar/game/Cell.scala#L109).